sky/jobs/utils.py

```diff
@@ -274,11 +270,19 @@ def _handle_legacy_job(job_id: int):
             failure_reason = ('Inconsistent internal job state. This is a bug.')
         elif pid is None:
             # Non-legacy job and controller process has not yet started.
-            if schedule_state in (
+            controller_status = job_lib.get_status(job_id)
+            if controller_status in [
+                    job_lib.JobStatus.FAILED_SETUP,
+                    job_lib.JobStatus.FAILED_DRIVER, job_lib.JobStatus.FAILED
+            ]:
+                # We should fail the case where the controller status is
+                # failed, as it is likely due to the job for submitting the
+                # managed job to scheduler failed.
+                logger.error('Failed to submit the managed job to scheduler.')
```

Comment:
@cg505, I added this so that jobs failed before scheduler is triggered for the job will be marked as FAILED_CONTROLLER. Wdyt?

Reply:
A bit concerned here, since the job_lib status could be failed but the managed job was successfully submitted, then something else failed after. In that case we could race with the scheduler and get into some weird undefined behavior.
If we want to add this check, we should also verify that the schedule_state is INACTIVE (submit to scheduler failed), and we need to check schedule_state _after_ we check the job_id status. Otherwise, if the job is WAITING or some other schedule_state, we need to make sure to stop the controller process.

Reply:
I see, I can add an additional condition for checking the `schedule_state` to be `INACTIVE`. I was assuming the job_lib status should always be `SUCCEEDED` if the managed job was successfully submited.

Reply:
There is also a question of what we want to do if the job_lib status is some FAILED state but the job was successfully submitted (schedule_state is not INACTIVE). This seems like a problem, so FAILED_CONTROLLER is reasonable. We just have to be careful about how to do it:
- set job to DONE
- get controller process PID
- if not None, kill controller process and children
- clean up cluster

or something like that

Reply:
I am not sure in what case it could happen that the `job_lib` can fail when the job is current submitted to scheduler, since the job controller's run section seems only inlcudes the submission to scheduler?

Reply:
e.g. some ray issue out of our control

Reply:
Or even in the submission, `set_waiting` succeeds but `maybe_schedule_next_jobs` crashes for some reason
https://github.com/skypilot-org/skypilot/blob/54fe787d1fb31687cc78eb307db59d54c2d79076/sky/jobs/scheduler.py#L186-L188